lib/vm/src/instance/mod.rs

```diff
@@ -418,6 +423,23 @@ impl Instance {
         unsafe { self.vmctx_plus_offset(self.offsets.vmctx_stack_limit_begin()) }
     }
 
+    fn write_perf_profiler_map(&self) {
+        let pid = process::id();
+        let filename = format!("/tmp/perf-{}.map", pid);
+        let mut file = File::create(filename).expect("Unable to create file");
```

Comment:
Wrapping this file into a `BufWriter` would be a good idea.

Also, I believe these should be returning a `Result::Err` rather than panicking.

Reply:
Correct.

Reply:
Guess it doesn't belong to VM.